Theme border settings change + updating styles

toger5 opened the pull request:
Two commits:
 - one fixes that the border settings options does not remove borders in places where they are always necessary (for example pop up) which made border 0 not usable.

![noborderpopup](https://user-images.githubusercontent.com/16718859/28625423-8284dc86-721c-11e7-9103-417fdffe7035.png)

 - adds a an empty style for editor help focus so that the editor does not use the defualt theme (old theme)

![nooldthemeborderhelp](https://user-images.githubusercontent.com/16718859/28625422-827bb7a0-721c-11e7-9c13-065e9cfc0db3.png)
 - the other one fixes updating on theme changes. for help background + script panel to prevent these effects:

![script panel ncolour update](https://user-images.githubusercontent.com/16718859/28625421-827ada92-721c-11e7-97d3-e88a245095f0.PNG)

toger5 committed:
border settings doesn't remove necessary borders anymore

toger5 committed:
editor: update helps and script panel on theme could change.

akien-mga commented:
cc @djrm @volzhs

djrm commented:
@toger5 Now that i think about it, having 0 border size does not make much sense until we can use shadows or something, i would rather remove the 0 border option.

toger5 commented:
@djrm personally I really like the look of it. And use it. just in some areas It did look odd. now it is usable and looks better (for my taste/ colour setup)

djrm commented:
@toger5 well, in that case i think its ok, later on we could add shadows to improve the overall look.

toger5 commented:
@djrm shadows are a great idea. My style box pr just needs to be merged ;)

djrm commented on editor/plugins/script_editor_plugin.cpp:
```diff
@@ -211,9 +211,11 @@ void ScriptEditorQuickOpen::_confirmed() {
 void ScriptEditorQuickOpen::_notification(int p_what) {
 
-	if (p_what == NOTIFICATION_ENTER_TREE) {
+	switch (p_what) {
```
why the switch

akien-mga replied:
That's the preferred style for parsing notifications.

akien-mga commented on editor/editor_themes.cpp:
```diff
@@ -143,7 +143,8 @@ Ref<Theme> create_editor_theme() {
 
-	const int border_width = (border_size % 3) * EDSCALE;
+	const int optional_border_width = (border_size % 3) * EDSCALE;
+	const int border_width = optional_border_width == 0 ? EDSCALE : optional_border_width;
```
The names don't make much sense to me. `border_width` in only used for two widgets, and `optional_border_width` for the others; in what way is it "optional"?

Also the logic looks a bit weird, `border_size == 3` or `6` or `9` would mean no `optional_border_width`? And actually border_size in [0, 1, 3, 4, 6, 7, 9, 10, etc.] would all result in the same behaviour... I might be missing something but it looks like a convoluted way to have a random distribution with 2/3 of probabilities for one value and 1/3 for the other :)

toger5 replied:
I reused the idea of limiting the border with to a maximum of 2 by using %. Thinking of it it might be better to just add another MAX() function...
the idea with border_width is, that it always is bigger than 1. (more precisely it always is at least EDSCALE). optional_border_width can be 0 (that is what I tried to describe with 'optional')

would you prefer when the inline if statement would be used directly in the two cases where border_width is used? than the confusing name optional_border_width is not necessary.

akien-mga commented on editor/editor_help.cpp:
```diff
@@ -1621,12 +1621,18 @@ void EditorHelp::_notification(int p_what) {
 			//forward->set_icon(get_icon("Forward","EditorIcons"));
 			//back->set_icon(get_icon("Back","EditorIcons"));
 			_update_doc();
+		}
```
You should `break;` after each case.

akien-mga commented on editor/editor_help.cpp:
```diff
@@ -1621,12 +1621,18 @@ void EditorHelp::_notification(int p_what) {
+		}
 
-		} break;
+		default: break;
```
That's not necessary.

akien-mga commented on editor/plugins/script_editor_plugin.cpp:
```diff
@@ -1064,58 +1066,73 @@ void ScriptEditor::_tab_changed(int p_which) {
+			add_style_override("panel", sb);
+		} break;
+
+		default:
```
I don't think that's necessary.

bojidar-bg replied:
Doesn't it silence some of the clang warnings though?

toger5 commented:
- I fixed the missing breaks
 - changed the logic to force borders to: ` MAX(EDSCALE, border_width) `. there is no optional border with anymore, border width can be 0. But for some UI elements it will now always draw a border.

djrm commented on editor/editor_themes.cpp:
```diff
@@ -143,7 +143,7 @@ Ref<Theme> create_editor_theme() {
 	Color light_color_2 = base_color.linear_interpolate(Color(1, 1, 1, 1), contrast * 1.5);
 
-	const int border_width = (border_size % 3) * EDSCALE;
+	const int border_width = MIN(border_size, 3) * EDSCALE;
```
Not exactly the same, since the % also works on negatives, but whatever, i dont think people will put a negative border size in their conf files.

toger5 replied:
Yea modulo also has its advantages... I chnged it, otherwise ppl would be confused when using values like: 7
If u want to we can change it to: `MIN(abs(border_size)), 3)`
I also increased the max bider size from 2*EDSCLAE to 3*EDSCALE. 
Is that fine?

volzhs replied:
I wondered why modular used at the very first time.
It will make confuse. what about clamping it instead of just MIN?

toger5 commented on editor/editor_themes.cpp:
```diff
@@ -143,7 +143,7 @@ Ref<Theme> create_editor_theme() {
 	Color light_color_2 = base_color.linear_interpolate(Color(1, 1, 1, 1), contrast * 1.5);
 
-	const int border_width = (border_size % 3) * EDSCALE;
+	const int border_width = CLAMP(border_size, 0, 3) * EDSCALE;
```
using CLAMP now
